pulsar-broker/src/main/java/com/yahoo/pulsar/broker/web/WebService.java

```diff
@@ -63,52 +62,61 @@
+
+    private final PulsarService pulsar;
+    private final Server server;
+    private final List<Handler> handlers;
+    private final ExecutorService webServiceExecutor;
 
-    public WebService(ServiceConfiguration config, PulsarService pulsar) throws PulsarServerException {
-        this.config = config;
+    public WebService(PulsarService pulsar) throws PulsarServerException {
+        this.handlers = Lists.newArrayList();
         this.pulsar = pulsar;
-        this.webServiceExecutor = Executors.newFixedThreadPool(32, new DefaultThreadFactory("pulsar-web"));
+        this.webServiceExecutor = Executors.newFixedThreadPool(WebService.NUM_ACCEPTORS, new DefaultThreadFactory("pulsar-web"));
         this.server = new Server(new ExecutorThreadPool(webServiceExecutor));
         List<ServerConnector> connectors = new ArrayList<>();
 
         ServerConnector connector = new PulsarServerConnector(server, 1, 1);
-        connector.setPort(config.getWebServicePort());
+        connector.setPort(pulsar.getConfiguration().getWebServicePort());
         connector.setHost(pulsar.getBindAddress());
         connectors.add(connector);
 
-        if (config.isTlsEnabled()) {
+        if (pulsar.getConfiguration().isTlsEnabled()) {
             SslContextFactory sslCtxFactory = new SslContextFactory();
 
             try {
-                SSLContext sslCtx = SecurityUtility.createSslContext(config.isTlsAllowInsecureConnection(),
-                        config.getTlsTrustCertsFilePath(), config.getTlsCertificateFilePath(),
-                        config.getTlsKeyFilePath());
-                sslCtxFactory.setSslContext(sslCtx);
+                sslCtxFactory.setSslContext(
+                        SecurityUtility.createSslContext(
+                            pulsar.getConfiguration().isTlsAllowInsecureConnection(),
+                            pulsar.getConfiguration().getTlsTrustCertsFilePath(),
+                            pulsar.getConfiguration().getTlsCertificateFilePath(),
+                            pulsar.getConfiguration().getTlsKeyFilePath()));
             } catch (GeneralSecurityException e) {
                 throw new PulsarServerException(e);
             }
 
             sslCtxFactory.setWantClientAuth(true);
             ServerConnector tlsConnector = new PulsarServerConnector(server, 1, 1, sslCtxFactory);
-            tlsConnector.setPort(config.getWebServicePortTls());
+            tlsConnector.setPort(pulsar.getConfiguration().getWebServicePortTls());
             tlsConnector.setHost(pulsar.getBindAddress());
             connectors.add(tlsConnector);
         }
 
         // Limit number of concurrent HTTP connections to avoid getting out of file descriptors
-        connectors.stream().forEach(c -> c.setAcceptQueueSize(1024 / connectors.size()));
+        connectors.stream().forEach(c -> c.setAcceptQueueSize(WebService.MAX_CONCURRENT_REQUESTES / connectors.size()));
```

Comment:
While in here... is there a good reason for `connectors.stream().forEach` vs. just using `connectors.forEach`